### PR TITLE
OC checker

### PR DESCRIPTION
**Pull request by csanadtelbisz**
This PR contains a generic ordering consistency checker for concurrent systems with a proof-of-concept implementation for sequential C programs in the xcfa subproject.

(Merge #258 first.)

### CODE REVIEW

**Comment by leventeBajczi** [comment]
Does this have anything of value now that the c-frontend-fix has been merged?

**Comment by mondokm** [comment]
> Does this have anything of value now that the c-frontend-fix has been merged?

@csanadtelbisz can we close this PR?

**Comment by csanadtelbisz** [comment]
This PR adds pointer support with some other fixes to the xcfa OC checker. We should merge it.
(Newest features are added to the [xcfa-oc-validator](https://github.com/csanadtelbisz/theta/tree/xcfa-oc-validator) branch though. But we can wait a bit with merging those.)

File: subprojects/xcfa/c2xcfa/src/main/java/hu/bme/mit/theta/c2xcfa/FrontendXcfaBuilder.kt [hunk 1]
```diff
@@ -50,812 +50,997 @@ import hu.bme.mit.theta.frontend.transformation.model.types.complex.integer.CInt
 import hu.bme.mit.theta.frontend.transformation.model.types.simple.CSimpleTypeFactory
 import hu.bme.mit.theta.xcfa.model.*
 import hu.bme.mit.theta.xcfa.passes.CPasses
-import org.abego.treelayout.internal.util.Contract.checkState
 import java.math.BigInteger
 import java.util.stream.Collectors
+import org.abego.treelayout.internal.util.Contract.checkState
```
Comment (leventeBajczi) [inline subprojects/xcfa/c2xcfa/src/main/java/hu/bme/mit/theta/c2xcfa/FrontendXcfaBuilder.kt:55]: This is the wrong import

Comment (leventeBajczi) [reply subprojects/xcfa/c2xcfa/src/main/java/hu/bme/mit/theta/c2xcfa/FrontendXcfaBuilder.kt:55]: I think this was already wrong, but we should fix it.

**Comment by leventeBajczi** [comment]
Tests are failing and that should be fixed (I cannot see at a glance what the problem might be). Otherwise, I think this PR is a great addition to Theta, and we should merge it as soon as possible.

**Comment by leventeBajczi** [comment]
I got tired of re-checking the same reformatted files. The changes of this PR can be seen under [this](https://github.com/ftsrg/theta/compare/master-format..csanadtelbisz:theta:xcfa-oc) diff.

**Pull request merged**